File: C_Sharp_code/Example/WindowsFormsApp/CS_demo20/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace CS_demo20
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            Lion lion = new Lion();
            Console.WriteLine("name:"+lion.Name);

            Animal lion1;
            lion1 = lion;//向上隱含轉型父類別
            lion1.talk();//override 方法
        }

        private void button2_Click(object sender, EventArgs e)
        {
            Bike bike = new Bike();
            Console.WriteLine("name: "+bike.Name);
            Console.WriteLine("----------");
            Car bike1;
            bike1 = bike;
            bike1.display();//Bike class方法遮蔽後再去呼叫會是轉型後的父類方法
        }

        private void button3_Click(object sender, EventArgs e)
        {
            Apple apple = new Apple("Apple");
            MessageBox.Show(apple.ToString());
            Apple apple2 = new Apple("Apples");
            if (apple.Equals(apple2))
                MessageBox.Show("Equals");
            else
                MessageBox.Show("Not Equals");
        }
    }
}

File: C_Sharp_code/Example/WindowsFormsApp/VB_demo21/Form1.vb
Public Class Form1
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Dim horse As New Horse()
        Console.WriteLine("name:" & horse.Name)

        Dim horse1 As Animal
        horse1 = horse '向上隱含轉型父類別
        horse1.Talk() 'override 方法
    End Sub

    Private Sub button2_Click(sender As Object, e As EventArgs) Handles button2.Click
        Dim bike As New Bike()
        Console.WriteLine("name: " & bike.Name)
        Console.WriteLine("---------")
        Dim bike1 As Car
        bike1 = bike
        bike1.Display() 'Bike class方法遮蔽後再去呼叫會是轉型後的父類方法
    End Sub

    Private Sub button3_Click(sender As Object, e As EventArgs) Handles button3.Click
        Dim apple As New Apple("Apple")
        MessageBox.Show(apple.ToString())
        Dim apple2 As New Apple("Apple")

        'apple = apple2
        If (apple.Equals(apple2)) Then
            MessageBox.Show("Equals")
        Else
            MessageBox.Show("Not Equals")
        End If
    End Sub
End Class
